### PR TITLE
Issue 245: Cleaning enw_formula_as_data_list() (#245)

### PR DESCRIPTION
**Pull request by Lnrivas**
Cleaned enw_formula_as_data_list() as requested in task #245 , please check that it works to your liking so it can be merged asap :)

### CODE REVIEW

**Comment by pearsonca** [comment]
To get rid of the linting warnings: try running `lintr::lint_package()` locally, then dealing with the issues (at least those in the code you wrote), re-running, etc until nothing arises from your code.

**Comment by pearsonca** [comment]
The PR process is also to do changes against the `develop` branch, not `main`; see: https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md#development

File: R/model-tools.R [hunk 1]
```diff
@@ -31,45 +31,28 @@
 #' enw_formula_as_data_list(prefix = "missing")
 enw_formula_as_data_list <- function(formula, prefix,
                                      drop_intercept = FALSE) {
-  paste_lab <- function(string, lab = prefix) {
-    paste0(lab, "_", string)
-  }
+  data <- list()
+  data[c("fintercept", "fdesign", "fnrow", "findex", "fnindex", "fncol", "rdesign", "rncol")] <- 0
+  data[c("fdesign", "fnrow", "findex", "fnindex", "fncol", "rdesign", "rncol")] <- list(numeric())
```
Comment (pearsonca) [inline R/model-tools.R:36]: this seems to be the incorrect collection of columns to assign to `numeric()` - copy-paste error?

**Review by pearsonca — CHANGES_REQUESTED** [review]
This is a useful start, but as per comments:

 - please fix the linting issues
 - there also appears to be an assignment error. Did this pass tests? If so, seems like we also need to update the test conditions (to deal with the do-nothing-path).

Thanks for the work so far; looking forward to the update.

File: R/model-tools.R (hunk 1 above)
Comment (pearsonca) [reply R/model-tools.R:36]: Also the wrong columns to 0 - this is on me, for not being clearer in the issue framing.

**Comment by seabbs** [comment]
> The right amount of to-the-point or too clever / terse?

I think this might be the right amount? Thoughts @Lnrivas?

**Comment by Lnrivas** [comment]
Thanks for all the feedback. I will close this pr and make a new one as instructed to correctly work with develop branch and make all changes there.

**Pull request closed without merging**